main.go

```diff
@@ -157,56 +187,76 @@ func main() {
 	config.SetValue(entryPoint, *defaultEntryPoint)
 	config.SetDefaultValues()
 
-	var fatalError bool
-	for _, err := range config.Validate() {
-		switch err := err.(type) {
-		case ConfigWarning:
-			fmt.Fprintln(os.Stderr, warningString("%v", err))
-		case VersionMistmatchError:
-			fmt.Fprintln(os.Stderr, errorString("%v", err))
-			if *imageVersion == "-" {
-				// We consider this as a fatal error only if the version has not been explicitly specified on the command line
-				fatalError = true
-			}
-		default:
-			fmt.Fprintln(os.Stderr, errorString("%v", err))
-			fatalError = true
-		}
-	}
-	if fatalError {
+	if !validateVersion(*imageVersion) {
 		os.Exit(1)
 	}
 
 	if *getCurrentVersion {
-		fmt.Printf("tgf v%s\n", version)
+		Printf("tgf v%s\n", version)
 		os.Exit(0)
 	}
 
 	if *getAllVersions {
 		if filepath.Base(config.EntryPoint) != "terragrunt" {
-			fmt.Fprintln(os.Stderr, errorString("--all-version works only with terragrunt as the entrypoint"))
+			printErr(("--all-version works only with terragrunt as the entrypoint"))
 			os.Exit(1)
 		}
-		fmt.Println("TGF version", version)
+		Println("TGF version", version)
 		unmanaged = []string{"get-versions"}
 	}
 
-	if config.ImageVersion == nil && lastRefresh(config.GetImageName()) > config.Refresh || !checkImage(config.GetImageName()) || refresh {
-		refreshImage(config.GetImageName())
+	imageName := config.GetImageName()
+	if config.ImageVersion == nil && lastRefresh(imageName) > config.Refresh || !checkImage(imageName) || refresh {
+		refreshImage(imageName)
 	}
 
 	if *loggingLevel != "" {
 		config.LogLevel = *loggingLevel
 	}
 
-	if config.EntryPoint == "terragrunt" && unmanaged == nil && !debug && !getImageName {
+	if *pruneImages {
+		prune(config.Image)
+		os.Exit(0)
+	}
+
+	if config.EntryPoint == "terragrunt" && unmanaged == nil && !debugMode && !getImageName {
 		title := color.New(color.FgYellow, color.Underline).SprintFunc()
-		fmt.Println(title("\nTGF Usage\n"))
+		ErrPrintln(title("\nTGF Usage\n"))
 		app.Usage(nil)
 	}
 
+	if config.ImageVersion == nil {
+		actualVersion := GetActualImageVersion()
+		config.ImageVersion = &actualVersion
+		if !validateVersion(*imageVersion) {
+			os.Exit(2)
+		}
+	}
+
 	os.Exit(callDocker(unmanaged...))
 }
 
+func validateVersion(version string) bool {
+	for _, err := range config.Validate() {
+		switch err := err.(type) {
+		case ConfigWarning:
+			warning("%v", err)
+		case VersionMistmatchError:
+			printErr("%v", err)
+			if version == "-" {
+				// We consider this as a fatal error only if the version has not been explicitly specified on the command line
+				return false
+			}
+		default:
+			printErr("%v", err)
+			return false
+		}
+	}
+	return true
+}
+
+func printErr(format string, args ...interface{}) { ErrPrintln(errorString(format, args...)) }
+func warning(format string, args ...interface{})  { ErrPrintln(warningString(format, args...)) }
```

Comment:
to be consistent, it could be printWarning

Reply:
Done, changed printErr to printError to be consistent too!